osquery/tables/system/windows/registry.cpp

```diff
@@ -151,14 +166,16 @@ void queryKey(const std::string& keyPath, QueryData& results) {
     return;
   }
 
-  BYTE* bpDataBuff = new BYTE[cbMaxValueData];
   DWORD cchValue = maxKeyLength;
   TCHAR achValue[maxValueName];
 
   // Process registry values
   for (size_t i = 0, retCode = ERROR_SUCCESS; i < cValues; i++) {
+    DWORD lpData = (cbMaxValueData == 0) ? NULL : cbMaxValueData;
+    DWORD lpType;
+    BYTE* bpDataBuff =
+        (cbMaxValueData == 0) ? NULL : new BYTE[cbMaxValueData]();
     size_t cnt = 0;
-    ZeroMemory(bpDataBuff, cbMaxValueData);
```

Comment:
The 'new' ()' initializer for bpDataBuff, according to the c++11 spec, should initialize the array to all 0s:

```
    BYTE* bpDataBuff =
        (cbMaxValueData == 0) ? nullptr : new BYTE[cbMaxValueData]();
```

Since this gets re-inialized each loop, the ZeroMemory is uncessary now. Also no point in calling ZeroMemory if it's a nullptr.

See 5.3.4  New: http://cs.nyu.edu/courses/fall11/CSCI-GA.2110-003/documents/c++2003std.pdf